Sources/SwiftUINavigationCore/Alert.swift

```diff
@@ -0,0 +1,386 @@
+    ///   - actions: A view builder returning the alert's actions given the current alert state.
+    @available(iOS 15, macOS 12, tvOS 15, watchOS 8, *)
+    public func alert<Item, A: View>(
+      item: Binding<Item?>,
+      title: (Item) -> Text,
+      @ViewBuilder actions: (Item) -> A
+    ) -> some View {
+      alert(
+        item.wrappedValue.map(title) ?? Text(verbatim: ""),
+        isPresented: item.isPresent(),
+        presenting: item.wrappedValue,
+        actions: actions
+      )
+    }
+  }
+
+  // MARK: - Alert with static localized title
+
+  extension View {
+    /// Presents an alert from a binding to an optional value with a static title.
+    ///
+    /// This function is similar to the other `alert` overload, but it takes a `LocalizedStringKey` for
+    /// a static title instead of a dynamic one. This is useful when the title of the alert does not
+    /// depend on the item that is being presented.
+    ///
+    /// The alert is presented when the `item` binding is non-`nil`. The `actions` and `message`
+    /// closures receive the unwrapped item to customize the alert's actions and message. When the user
+    /// dismisses the alert by tapping one of its actions, the system sets the `item` binding to `nil`.
+    ///
+    /// Usage example:
+    ///
+    /// ```swift
+    /// struct AlertDemo: View {
+    ///   @State var randomMovie: Movie?
+    ///
+    ///   var body: some View {
+    ///     Button("Pick a random movie", action: self.getRandomMovie)
+    ///       .alert("Random Movie", item: self.$randomMovie) { _ in
+    ///          Button("Pick another", action: self.getRandomMovie)
+    ///          Button("I'm done", action: self.clearRandomMovie)
+    ///        } message: {
+    ///          Text($0.summary)
+    ///        }
+    ///   }
+    ///
+    ///   func getRandomMovie() {
+    ///     self.randomMovie = Movie.allCases.randomElement()
+    ///   }
+    ///
+    ///   func clearRandomMovie() {
+    ///     self.randomMovie = nil
+    ///   }
+    /// }
+    /// ```
+    ///
+    /// - Parameters:
+    ///   - titleKey: A `LocalizedStringKey` representing the static title of the alert.
+    ///   - item: A binding to an optional value that determines whether an alert should be presented.
+    ///     When the binding is updated with a non-`nil` value, it is unwrapped and passed to the
+    ///     modifier's closures. You can use this data to populate the fields of an alert that the
+    ///     system displays to the user. When the user presses or taps one of the alert's actions, the
+    ///     system sets this value to `nil` and dismisses the alert.
+    ///   - actions: A view builder returning the alert's actions given the current alert state.
+    ///   - message: A view builder returning the message for the alert given the current alert state.
+    @available(iOS 15, macOS 12, tvOS 15, watchOS 8, *)
+    public func alert<Item, A: View, M: View>(
+      _ titleKey: LocalizedStringKey,
```

Comment:
Let's not worry about static titles for now. If folks want they can either ignore the unwrapped item, or they can use one of Apple's messier vanilla APIs.